### ISSUE TITLE
Horizon deprecations tags

### ISSUE THREAD

**Issue by bartekn** [stellar/go#833]
It's very easy to forget about removing deprecated fields/methods in Horizon. We should do it automatically.

I propose the following comment syntax:
```go
// Deprecated - remove in: horizon-v[version]
```
Usage example:
```go
type Root struct {
	Links struct {
		Account             hal.Link  `json:"account"`			Account             hal.Link  `json:"account"`
		AccountTransactions hal.Link  `json:"account_transactions"`			AccountTransactions hal.Link  `json:"account_transactions"`
		Assets              hal.Link  `json:"assets"`			Assets              hal.Link  `json:"assets"`
		Friendbot           *hal.Link `json:"friendbot,omitempty"`			Friendbot           *hal.Link `json:"friendbot,omitempty"`
		Metrics             hal.Link  `json:"metrics"`			Metrics             hal.Link  `json:"metrics"`
		OrderBook           hal.Link  `json:"order_book"`			OrderBook           hal.Link  `json:"order_book"`
		Self                hal.Link  `json:"self"`			Self                hal.Link  `json:"self"`
		Transaction         hal.Link  `json:"transaction"`			Transaction         hal.Link  `json:"transaction"`
		Transactions        hal.Link  `json:"transactions"`			Transactions        hal.Link  `json:"transactions"`
	} `json:"_links"`		} `json:"_links"`


 	HorizonVersion       string `json:"horizon_version"`		HorizonVersion               string `json:"horizon_version"`
	StellarCoreVersion   string `json:"core_version"`		StellarCoreVersion           string `json:"core_version"`
	HorizonSequence      int32  `json:"history_latest_ledger"`		HorizonSequence              int32  `json:"history_latest_ledger"`
	HistoryElderSequence int32  `json:"history_elder_ledger"`		HistoryElderSequence         int32  `json:"history_elder_ledger"`
	CoreSequence         int32  `json:"core_latest_ledger"`		CoreSequence                 int32  `json:"core_latest_ledger"`
	NetworkPassphrase    string `json:"network_passphrase"`		NetworkPassphrase            string `json:"network_passphrase"`
	// Deprecated - remove in: horizon-v0.17.0
	ProtocolVersion      int32  `json:"protocol_version"`		ProtocolVersion              int32  `json:"protocol_version"`
	CurrentProtocolVersion       int32  `json:"current_protocol_version"`
	CoreSupportedProtocolVersion int32  `json:"core_supported_protocol_version"`
}	}
```
Then we can create a simple `grep`-like script that searches for such tags in `/services/horizon/*` and `/protocols/horizon/*` files. If the current build is a release build, the build should error if there are any tags found related to the current version that should be removed.

**Comment by ire-and-curses** [comment]
This sounds good to me.

**Comment by bartekn** [comment]
Closed in #838.
